src/main/java/org/jvnet/hudson/test/InboundAgentRule.java

```diff
@@ -465,7 +481,7 @@ private static AgentArguments getAgentArguments(JenkinsRule r, String name) thro
         if (!launcher.getWorkDirSettings().isDisabled()) {
             commandLineArgs = launcher.getWorkDirSettings().toCommandLineArgs(c);
         }
-        File agentJar = new File(r.jenkins.getRootDir(), "agent.jar");
+        File agentJar = new File(System.getProperty("java.io.tmpdir"), "agent.jar");
```

Comment:
this appears to completely break some things:

e.g. `-DforkCount=<number_greater_than_one>` as the file may not exist for 2 forks ([which use the same surefire temp director](https://github.com/jenkinsci/plugin-pom/blob/55de1268f3ffe29a657dc7b233c676697f2a1c4d/pom.xml#L806)y) , is completely written by one and the other and then used whilst the other is overwriting it.

2. the ability to quickly switch between version using `-DJenkins.version=x` (you now need to `mvn clean` as opposed to just using the incremental compilation which is faster)

Reply:
Yeah a temp file would be better for this case. I can prepare a patch for it.